src/lightning_app/cli/cmd_apps.py

```diff
@@ -0,0 +1,81 @@
+import json
+from datetime import datetime
+
+from lightning_cloud.openapi import Externalv1LightningappInstance, V1LightningappInstanceState
+from rich.console import Console
+from rich.table import Table
+from rich.text import Text
+
+from lightning_app.cli.core import Formatable
+from lightning_app.utilities.cloud import _get_project
+from lightning_app.utilities.network import LightningClient
+
+
+class AppManager:
+    """AppManager implements API calls specific to Lightning AI BYOC apps."""
+
+    def __init__(self):
+        self.api_client = LightningClient()
+
+    def list(self, cluster_id=None):
+        project = _get_project(self.api_client)
+
+        args = {
+            "project_id": project.project_id,
+        }
+        if cluster_id is not None:
+            args["cluster_id"] = cluster_id
+
+        resp = self.api_client.lightningapp_instance_service_list_lightningapp_instances(**args)
+        apps = resp.lightningapps
+        while resp.next_page_token is not None and resp.next_page_token != "":
+            args["page_token"] = resp.next_page_token
+            resp = self.api_client.lightningapp_instance_service_list_lightningapp_instances(**args)
+            apps = apps + resp.lightningapps
+        console = Console()
+        console.print(AppList(resp.lightningapps).as_table())
+
+
+class AppList(Formatable):
+    def __init__(self, apps: [Externalv1LightningappInstance]):
+        self.apps = apps
+
+    def as_json(self) -> str:
+        return json.dumps(self.apps)
+
+    def as_table(self) -> Table:
+        table = Table("id", "name", "status", "cluster", "created", show_header=True, header_style="bold green")
+        phases = {
```

Comment:
nit: we should probably do a bit better than just one to one mapping, like the "terminating" one we did below for other statuses, like "restarting". Probably looking at the FE codebase is a good reference?

Reply:
let me see what other transitions we have in there and add them below :+1:

Reply:
added other state transitions like "not yet run" and "restarting" :+1: